pandas/core/arrays/boolean.py

```diff
@@ -59,6 +59,7 @@ class BooleanDtype(ExtensionDtype):
     >>> pd.BooleanDtype()
     BooleanDtype
     """
+    name = "boolean"
```

Comment:
changed for consistency with IntervalDType and CatergoricalDType. ExtensionDtype subclasses with class attribute for name do not need to override construct_from_string

on master:

```python
>>> import pandas as pd; pd.__version__
'0.26.0.dev0+1554.g0913ed04d'
>>>
>>> pd.core.dtypes.dtypes.CategoricalDtype.name
'category'
>>>
>>> pd.core.dtypes.dtypes.IntervalDtype.name
'interval'
>>>
>>> pd.core.arrays.boolean.BooleanDtype.name
<property object at 0x0000026B6A278E58>
>>>
>>> pd.core.arrays.string_.StringDtype.name
<property object at 0x0000026B6A550908>
>>>
```

Reply:
Is it possible to do this the other way? I think property is a little nicer than class attributes

@TomAugspurger

Reply:
`construct_from_string` is a class method, so I think it makes more sense to make `name` a class attribute. 

(should probably consider making PeriodDType.name and DatetimeTZDtype.name class attributes for consistency)

Reply:
I guess the only downside is non-dynamic names? I don't think we can make `PeriodDtype.name` a class attribute, since it depends on the freq.

Reply:
for the static ones this seems fine